### PR TITLE
Receive Screen

### PR DESCRIPTION
**Pull request by 0xLe0x**
- Receive(Request) Screens
- wrapped with `TransferProvider` for sharing states between screens

### CODE REVIEW

File: packages/app/assets/img/qr.png
Comment (0xBigBoss) [inline packages/app/assets/img/qr.png:1]: let's drop this, just make it red color or something instead and leave a TODO in the code.

File: packages/app/features/send/components/modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export * from './send-confirm-modal'
```
Comment (0xBigBoss) [inline packages/app/features/send/components/modal/index.tsx:1]: please no barrel exports. list out each export.

File: packages/app/features/send/providers/transfer-provider.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { createContext, useContext, useState } from 'react';
+import { ISharedStateContextType, ISharedStateProviderProps, ISharedStateType } from '../types';
+import { IconEthereum, IconUSDC } from 'app/components/icons';
+
+const SharedStateContext = createContext<ISharedStateContextType | undefined>(undefined);
+
+export const useSharedState = () => {
```
Comment (0xBigBoss) [inline packages/app/features/send/providers/transfer-provider.tsx:7]: rename to something more appropriate. `useTransferContext` e.g.

File: packages/app/features/send/providers/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from './transfer-provider'
```
Comment (0xBigBoss) [inline packages/app/features/send/providers/index.ts:1]: ditto barrel files

File: packages/app/features/send/screens/receive/index.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+export const ReceiveScreen = () => {
+  const [[currentScreen, direction], setCurrentScreen] = useState<[IReceiveScreenType, number]>(['receive-qrcode', -1]);
+
+  const ScreenComponent = screens[currentScreen];
+
+  return (
+    <TransferProvider>
+      <AnimationLayout currentKey={currentScreen} direction={direction}>
```
Comment (0xBigBoss) [inline packages/app/features/send/screens/receive/index.tsx:22]: love this animation layout so useful. animations are look great.

File: packages/app/features/send/screens/receive/receive-qrcode.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+          fontWeight={'700'}
+          mr={'$2.5'}
+          $shorter={{ fontSize: '$8' }}
+        >
+          Receive
+        </SizableText>
+      </XStack>
+
+      <YStack ai={'center'}>
+        <Image
```
Comment (0xBigBoss) [inline packages/app/features/send/screens/receive/receive-qrcode.tsx:45]: drop the qr code image for now, add a todo, just make it a red box for now or something.

File: packages/app/features/send/screens/send/send-it.tsx [hunk 6]
```diff
@@ -9,12 +9,11 @@ import {
...
 import { useState } from "react"
 import { Link } from '@my/ui/src/components'
-import { IconClose } from "app/components/icons/IconClose"
+import { IconBack, IconClose } from "app/components/icons"
 import { SendButton } from "app/components/layout/footer/components/SendButton"
-import { IconEthereum } from "app/components/icons"
-import { ConfirmModal } from "../../components/send"
-import { IconBack } from "app/components/icons/IconBack"
-import { SendScreenProps } from "../../types"
+import { SendConfirmModal } from "../../components/modal"
```
Comment (0xBigBoss) [inline packages/app/features/send/screens/send/send-it.tsx:14]: imo it's cleaner to just import using `app/components/modal`. we have the typescript paths setup for it.

File: packages/app/features/send/types.ts [hunk 7]
```diff
@@ -1,34 +1,60 @@
 import { ButtonProps } from "@my/ui"
 
-export type NumPadProps = {
+export type INumPadProps = {
```
Comment (0xBigBoss) [inline packages/app/features/send/types.ts:3]: I would change the `type` to an `interface` to match the name or just drop the `I` to be consistent with naming.

File: packages/app/features/send/types.ts [hunk 8]
```diff
@@ -1,34 +1,60 @@
...
+
+export type ISendScreenProps = {
+  setCurrentScreen: ([currentScreen, direction]: [currentScreen: ISendScreenType, direction: number]) => void
+}
+
+export type IReceiveScreenProps = {
+  setCurrentScreen: ([currentScreen, direction]: [currentScreen: IReceiveScreenType, direction: number]) => void
+}
+
+export type ISharedStateType = {
```
Comment (0xBigBoss) [inline packages/app/features/send/types.ts:42]: rename to something more appropriate e.g. TransferContext

**Review by 0xBigBoss — COMMENTED** [review]
looking pretty good. just a few minor things, looks good to merge after your review them.

**Review by 0xBigBoss — COMMENTED** [review]
when the window is too short, i think the vertical scrollbar causes the horizontal scrollbar. try including the tamagui reset css https://tamagui.dev/docs/core/configuration#including-the-reset-css

Screenshot 2023-12-17 at 15 44 17

i think once that is done, this one is good to merge. we still need to figure out the flows together since still not a fan of this flow for requesting payment.

**Comment by 0xLe0x** [comment]
> when the window is too short, i think the vertical scrollbar causes the horizontal scrollbar. try including the tamagui reset css https://tamagui.dev/docs/core/configuration#including-the-reset-css

tried reset css but it didn't help. Just set `overflow: hidden` on top of the screen

**Review by 0xBigBoss — APPROVED** [review]

**Pull request merged**